daisy/disk.go

```diff
@@ -131,8 +131,12 @@ func (d *Disk) validate(ctx context.Context, s *Step) DError {
 		if _, err := s.w.images.regUse(d.SourceImage, s); err != nil {
 			errs = addErrs(errs, Errf("%s: can't use image %q: %v", pre, d.SourceImage, err))
 		}
+	} else if d.SourceSnapshot != "" {
+		if _, err := s.w.snapshots.regUse(d.SourceSnapshot, s); err != nil {
+			errs = addErrs(errs, Errf("%s: can't use snapshot %q: %v", pre, d.SourceSnapshot, err))
+		}
 	} else if d.Disk.SizeGb == 0 {
-		errs = addErrs(errs, Errf("%s: SizeGb and SourceImage not set", pre))
+		errs = addErrs(errs, Errf("%s: SizeGb, SourceSnapshot or SourceImage not set", pre))
```

Comment:
What's the plan for when a user includes sourceSnapshot along with sourceImage and/or sizeGb?

Reply:
that is allowed: when creating a disk from image/snapshot, cuser can specify a larger disk size.

Reply:
ahh, cool